rs/replicated_state/src/canister_state/system_state.rs

```diff
@@ -334,32 +359,38 @@ impl TaskQueue {
 
     pub fn enqueue(&mut self, task: ExecutionTask) {
         match task {
+            ExecutionTask::AbortedInstallCode { .. }
+            | ExecutionTask::PausedExecution { .. }
+            | ExecutionTask::PausedInstallCode(_)
+            | ExecutionTask::AbortedExecution { .. } => {
+                debug_assert!(self.paused_or_aborted_task.is_none());
+                self.paused_or_aborted_task = Some(task);
+            }
             ExecutionTask::OnLowWasmMemory => {
                 self.on_low_wasm_memory_hook_status.update(true);
             }
-            ExecutionTask::Heartbeat
-            | ExecutionTask::GlobalTimer
-            | ExecutionTask::AbortedInstallCode { .. }
-            | ExecutionTask::PausedExecution { .. }
-            | ExecutionTask::PausedInstallCode(_)
-            | ExecutionTask::AbortedExecution { .. } => self.queue.push_front(task),
+            ExecutionTask::Heartbeat | ExecutionTask::GlobalTimer => self.queue.push_front(task),
         };
     }
 
     pub fn is_empty(&self) -> bool {
-        self.queue.is_empty()
+        self.paused_or_aborted_task.is_none() && self.queue.is_empty()
     }
 
     pub fn len(&self) -> usize {
-        self.queue.len()
+        self.paused_or_aborted_task.as_ref().map_or(0, |_| 1) + self.queue.len()
     }
 
     pub fn peek_hook_status(&self) -> OnLowWasmMemoryHookStatus {
         self.on_low_wasm_memory_hook_status
     }
 
-    pub fn get_queue(&self) -> &VecDeque<ExecutionTask> {
-        &self.queue
+    pub fn get_queue(&self) -> VecDeque<ExecutionTask> {
+        let mut queue = self.queue.clone();
+        if let Some(task) = self.paused_or_aborted_task.as_ref() {
+            queue.push_front(task.clone());
+        }
+        queue
```

Comment:
This will be removed in the follow-up https://dfinity.atlassian.net/browse/EXC-1752 when we introduce `CanisterStateBits` version of TaskQueue, so the conversion will be implicit.

Reply:
Done.